include/cluster.hpp

```diff
@@ -0,0 +1,108 @@
+
+#include <utility>
+#include <cmath>
+#include <stdexcept>
+
+class cluster
```

Comment:
```suggestion
class Cluster
```
as we discussed, or we do this change in an extra PR where we change junctions and breakends aswell

Reply:
Yes, I planned to create a separate PR to keep things tidy :)